### ISSUE TITLE
no-optional not working

### ISSUE THREAD

**Issue by PeterRock** [pnpm/pnpm#5900]
### pnpm version: 
7.23.0

### Code to reproduce the issue:
pnpm i --no-optional
### Expected behavior:
run success if optional package was installed failed

### Actual behavior:
Crashed, and install nothing;
There is no node_modules floder for fresh install

### Additional information:
node 14.21.2
macOS 12.3.1

I tried run 'npm i', it's fine and tips SKIPING OPTIONAL DEPENENCY

**Comment by zkochan** [comment]
Provide a repository that reproduces the issue.

**Comment by PeterRock** [comment]
sorry, it's a protected project.

**Comment by PeterRock** [comment]
I resolved this problem.
The installation process was blocked when downloading `gkt.tgz` which was depended by pm2. Npm v5+ will ignore this failed message because it's optional dependency. but pnpm seems not work like that.
Finally i downloaded gkt-1.0.0.tgz and start a server by anywhere.js, after that, add below code in package.json. I jumped over this problem.

```
"resolutions": {
  "gkt": "http://127.0.0.1:8000/gkt-1.0.0.tgz"
}
```
